market-test/src/test/java/com/axonivy/market/test/ValidateRepoTest.java

```diff
@@ -31,60 +63,35 @@ void everyProductHasAValidMetaJson()
   private void assertValidMetaJson(Path path)
   {
     var metaPath = path.resolve("meta.json");
-    var json = toJsonObject(metaPath);
-    JSONObjectAssert.assertThat(json, metaPath)
-            .requireStringPropertyWithLength("id", 5, 25)
-            .optionalStringPropertyWithPattern("version", "^(\\d+\\.)?(\\d+\\.)?(\\*|\\d+)$")
-            .requireStringPropertyWithLength("name", 4, 24)
-            .requireStringPropertyWithLength("description", 5, 200)
-            .optionalStringPropertyWithMinLength("vendor", 3)
-            .optionalStringPropertyWithPattern("compatibility", "^(\\d+\\.)?(\\d+\\.)?(\\*|\\d+)[+]?$")
-            .optionalStringPropertyWithPattern("platformReview", "^([0-4])?(\\.[5])?$|^5$")
-            .optionalStringPropertyWithMinLength("industry", 2)
-            .optionalStringPropertyWithFixedValues("cost", "paid") // free is default
-            .requireStringPropertyWithFixedValues("type", "connector", "solution", "process", "util")
-            .optionalBooleanProperty("listed", true)
-            .optionalStringPropertyWithFixedValues("versionDisplay", "portal")
-            .optionalStringArrayProperty("tags");
+    var json = load(metaPath);
 
     if (json.has("vendor")) {
       assertThat(json.has("vendorUrl")).as("if vendor is specified vendorUrl must be also specified").isTrue();
       assertThat(json.has("vendorImage")).as("if vendor is specified vendorImage must be also specified").isTrue();
     }
     if (json.has("vendorUrl")) {
-      var vendorUrl = json.getString("vendorUrl");
+      var vendorUrl = json.get("vendorUrl").asText();
       assertThat(checkUriExists(vendorUrl)).as("vendor url is not reachable " + vendorUrl).isTrue();
     }
     if (json.has("vendorImage")) {
-      var image = json.getString("vendorImage");
+      var image = json.get("vendorImage").asText();
       var vendorImage = path.resolve(image);
       assertThat(vendorImage).exists();
     }
 
     if (json.has("mavenArtifacts"))
     {
-      var array = json.getJSONArray("mavenArtifacts");
-      for (int i = 0; i < array.length(); i++)
-      {
-        var mavenArtifact = array.getJSONObject(i);
-        JSONObjectAssert.assertThat(mavenArtifact, metaPath)
-                .requireStringPropertyWithMinLength("name", 5)
-                .requireStringPropertyWithMinLength("groupId", 5)
-                .requireStringPropertyWithMinLength("artifactId", 5)
-                .optionalBooleanProperty("makesSenseAsMavenDependency", false)
-                .optionalStringPropertyWithMinLength("key", 5)
-                .optionalStringPropertyWithFixedValues("type", "zip", "nbm", "jar")
-                .optionalBooleanProperty("doc", false);
-
-        var groupId = mavenArtifact.getString("groupId");
-        var artifactId = mavenArtifact.getString("artifactId");
```

Comment:
it's all neatly documented in the schema now:
https://jenkins.ivyteam.io/job/core_json-schema/job/cc23.marketSchema/lastSuccessfulBuild/artifact/workspace/ch.ivyteam.ivy.market.schema/target/schema/market/0.0.1/meta.json